src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -610,13 +627,23 @@ final case class AgreementApiServiceImpl(
       .failed(AgreementActivationFailed(agreement.id))
       .whenA(failureStates.contains(newState))
 
+    def performActivation(seed: UpdateAgreementSeed, agreement: AgreementManagement.Agreement): Future[Unit] = for {
+      producer <- tenantManagementService.getTenant(agreement.producerId)
+      _        <- agreementContractCreator.create(
+        agreement = agreement,
+        eService = eService,
+        consumer = consumer,
+        producer = producer,
+        seed = seed
+      )
+      _        <- sendActivationEnvelope(agreement, producer, consumer, eService)
+    } yield ()
+
     for {
       seed    <- getUpdateSeed()
-      _       <-
-        if (firstActivation) agreementContractCreator.create(agreement, eService, consumer, seed)
-        else Future.unit
       updated <- agreementManagementService.updateAgreement(agreement.id, seed)
       _       <- failOnActivationFailure()
+      _       <- performActivation(seed, updated).whenA(firstActivation)
```

Comment:
I admin I'm a bit lost with current conventions, are none/all/some cats functions ok on `Future`?

Reply:
None of them is okay, they shouldn't be used